docs/Guide/application-commands/interaction-handlers/what-are-they.mdx

```diff
@@ -3,7 +3,180 @@ sidebar_position: 1
 title: What are they?
 ---
 
-- Basics
-- Folder structure (pieces related)
+These are interaction handlers! A simple class you can extend to handle almost all the interactions you may receive in your bot, in a simple, clean and abstract way.
 
-<!-- TODO: https://vladfrangu.notion.site/Application-Commands-3d1683f23f0a4fa6b3e682bc8c1cc1b9#ceb1a6f5ec154f85987a95181d512660 -->
+They are located in the `interaction-handlers` folder in your bot (just like how you most likely have a `commands` folder or a `listeners` folder). Same rules apply when it comes to loading and everything!
+
+## How does it work?
+
+When creating your interaction handler, you have to specify the type it is. Currently, there are three types:
+  
+| Enum Name  | What it will run for |
+| ------------- | ------------- |
+| Autocomplete | Autocomplete for interaction commands  |
+| Button | Button presses  |
+| ModalSubmit | Pop up modals  |
+| MessageComponent | components  | Any message component interactions received (right now buttons, select menus, or modals)  |
+| SelectMenu | Drop-down menus in messages  |
+
+These types define what the handler should get. For instance, specifying a `Button` interaction handler type will make that handler receive **just** button interactions.
+
+But that's not all! There's also a `parse` method which can be used for even more granular control (read below).
+
+:::warning
+
+By default, there is no filter for the `parse` method! This means you'll get all interactions of that type, and if you want to override that (for example to only run that handler for certain ids), you'll want to override the `parse` method in your code!
+
+:::
+
+## The `parse` method
+
+When you're creating handlers, you may want to run certain handlers only when the custom id matches something you want. In order to allow filtering, as well as optional additional pre-parsing of the custom id, we provide the `parse` method on handlers, which you should override with your own code. The return of the `parse` method **must** be a `Some` / `None`, which you can do with the provided shortcuts in the class: `this.some()` and `this.none()`.
+
+:::info
+
+Returning a some means that the handler's run method should be called. Returning a none means this handler should be skipped for that interaction. If you've used our Args system, this might seem familiar with the ok/err that is present there. 
```

Comment:
Link to args too

```suggestion
Returning a `some` means that the handler's `run` method should be called. Returning a `none` means this handler should be skipped for that interaction. If you've used our Args system, this might seem familiar with the `ok`/`err` that is present there. 
```